src/CMakeLists.txt

```diff
@@ -33,9 +33,10 @@ target_include_directories (fargparse PUBLIC
 target_include_directories (fargparse PRIVATE ${FARGPARSE_SOURCE_DIR}/include)
 add_library(FARGPARSE::fargparse ALIAS fargparse)
 
-include(build_submodule)
-build_submodule(../extern/gFTL-shared PROJECT GFTL_SHARED TARGET GFTL_SHARED::gftl-shared)
-
+if (NOT TARGET GFTL_SHARED::gftl-shared)
+  include(build_submodule)
+  build_submodule(../extern/gFTL-shared PROJECT GFTL_SHARED TARGET GFTL_SHARED::gftl-shared)
+endif ()
```

Comment:
In theory the build_submodule() macro should only be using the submodule if the specified project is not found.   How is this extra check outside the macro really changing anything? 

Even if it is necessary, could the conditional possibly be moved inside the `build_submodule` macro?

Reply:
You're correct. I didn't see this feature in `build_submodule()`. Thanks for the clarification.